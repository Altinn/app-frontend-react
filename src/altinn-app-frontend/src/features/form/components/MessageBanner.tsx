import * as React from 'react';
import { Grid, makeStyles } from '@material-ui/core';
import { AltinnAppTheme } from 'altinn-shared/theme';
import { getLanguageFromKey } from 'altinn-shared/utils';
import { ILanguage } from 'altinn-shared/types';
import classNames from 'classnames';

const useStyles = makeStyles({
  banner: {
    margin: '-36px -96px 36px -96px',
    padding: '10px 96px',
  },
  default: {
    backgroundColor: AltinnAppTheme.altinnPalette.primary.greyLight,
  },
  error: {
    backgroundColor: AltinnAppTheme.altinnPalette.primary.redLight,
  }
});

export interface IMessageBannerProps {
  language: ILanguage;
  error?: boolean;
  messageKey: string;
}

export default function MessageBanner(props: IMessageBannerProps) {
  const classes = useStyles();

  return (
    <Grid
      id={'MessageBanner-container'}
      item={true}
      className={classNames(
        classes.banner,
        props.error ? classes.error : classes.default
      )}
      data-testid={'MessageBanner-container'}
    >
      <span>{getLanguageFromKey(props.messageKey, props.language)}</span>
    </Grid>
  );
}
