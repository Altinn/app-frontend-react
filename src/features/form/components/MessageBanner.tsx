import React from 'react';

import classNames from 'classnames';

import css from 'src/features/form/components/MessageBanner.module.css';
import { getLanguageFromKey } from 'src/language/sharedLanguage';
import type { ILanguage } from 'src/types/shared';

interface IMessageBannerProps {
  language: ILanguage;
  error?: boolean;
  messageKey: string;
}

export const MessageBanner = ({ language, error, messageKey }: IMessageBannerProps) => (
  <div
    className={classNames(css.banner, error ? css.error : css.default)}
    data-testid='MessageBanner-container'
  >
    <span>{getLanguageFromKey(messageKey, language)}</span>
  </div>
);
