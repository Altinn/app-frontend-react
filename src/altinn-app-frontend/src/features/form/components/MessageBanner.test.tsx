import * as React from 'react';
import { render } from '@testing-library/react';
import { AltinnAppTheme } from 'altinn-shared/theme';

import MessageBanner from './MessageBanner';

describe('form/components/MessageBanner.tsx', () => {
  let mockLanguage: any;
  let mockMessageKey: string;

  beforeEach(() => {
    mockMessageKey = 'form_filler.required_description';
    mockLanguage = {
      'form_filler': {
        'required_description': 'Obligatoriske felter er merket med *',
      }
    }
  });

  test('form/components/MessageBanner -- should match snapshot', () => {
    const { asFragment } = render(
      <MessageBanner
        language={mockLanguage}
        messageKey={mockMessageKey}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('form/components/MessageBanner -- should have red background when error==true', () => {
    const { getByTestId } = render(
      <MessageBanner
        language={mockLanguage}
        messageKey={mockMessageKey}
        error={true}
      />
    );

    const messageBanner: HTMLElement = getByTestId('MessageBanner-container');
    expect(messageBanner).toBeInTheDocument();
    expect(messageBanner.className).toContain('error');
    const backgroundColor = window.getComputedStyle(messageBanner).backgroundColor;
    expect(backgroundColor).toEqual(convertToRgb(AltinnAppTheme.altinnPalette.primary.redLight));
  })
});

const convertToRgb = (hexValue: string): string => {
  const aRgbHex = hexValue.replace('#', '').match(/.{1,2}/g);
  const aRgb = [
      parseInt(aRgbHex[0], 16),
      parseInt(aRgbHex[1], 16),
      parseInt(aRgbHex[2], 16)
  ];
  return `rgb(${aRgb[0]}, ${aRgb[1]}, ${aRgb[2]})`;
}
