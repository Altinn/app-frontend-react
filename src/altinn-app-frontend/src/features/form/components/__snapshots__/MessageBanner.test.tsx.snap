// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form/components/MessageBanner.tsx form/components/MessageBanner -- should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-banner-1 makeStyles-default-2 MuiGrid-item"
    data-testid="MessageBanner-container"
    id="MessageBanner-container"
  >
    <span>
      Obligatoriske felter er merket med *
    </span>
  </div>
</DocumentFragment>
`;
